src/modules/audio/spatializer.h

```diff
@@ -2,14 +2,46 @@
 #include "core/maf.h"
 
 typedef struct {
-  bool (*init)(void);
+	int maxSourcesHint;
+	int fixedBuffer;
+	int sampleRate;
```

Comment:
Caution: #347 adds the ability to dynamically change the device sample rate from Lua.

It seems like either the spatializer would need to be able to reinitialize when this occurs, or the dynamic sample rate feature would need to be removed.

Note that SteamAudio also needs the sample rate specified and would run into this same issue.

Reply:
It would not be at all hard to shut down and restart the spatializer as long as it is okay for all sounds to potentially stop. In fact a good way to implement the sample rate change would be to call spatializer.destroy and then spatializer.init again. This would require zero change to the spatializer code (unless the audio callback is running during the sample rate change).

Reply:
Interesting, phonon only supports 24000, 44100, 48000.  Not immediately relevant though